### DISCUSSION TITLE
how to properly stop frankenphp and laravel octane as a standalone executable?

### DISCUSSION THREAD

**Discussion by tonychuuy** [php/frankenphp#963] [Q&A]
I'm using frankenphp binary to run laravel octane as the documentation says:

`PATH="$PWD:$PATH" ./frankenphp php-cli artisan octane:frankenphp`

basically I have supervisor with the following command: `frankenphp php-cli artisan octane:start`

```
[program:octane]
process_name=%(program_name)s_%(process_num)02d
directory=/var/www/site
command=frankenphp php-cli artisan octane:start
environment=PATH="/var/www/site:$PATH"
autostart=true
autorestart=true
stopasgroup=true
killasgroup=true
user=www-data
```

Whe the process starts with `sudo supervisorctl start all` it creates the following processes :

```
www-data    3371  4.6  3.0 1465728 242452 ?      Sl   20:25   0:01 frankenphp php-cli artisan octane:frankenphp
www-data    3389  3.7  3.1 1641008 248516 ?      Sl   20:25   0:00 /var/www/site/frankenphp run -c /tmp/frankenphp_8c7fc70e69b6cf3244d282de163ee95e/vendor/laravel/octane/src/Commands/stubs/Caddyfile

```

The issue is when I stop the processes using supervisor `sudo supervisorctl stop all`, it kills the first process,  the second one remains alive.

```
www-data    3389  1.6  3.1 1641008 248516 ?      Sl   20:25   0:00 /var/www/site/frankenphp run -c /tmp/frankenphp_8c7fc70e69b6cf3244d282de163ee95e/vendor/laravel/octane/src/Commands/stubs/Caddyfile
```

then I cannot start them again because the port 8000 is already in use. I suppose, I need to use something like octane:reload, but I needed to stop supervisor because I'm doing some setup and is not already finished.

**Comment by withinboredom** [comment]
It seems like there is an issue with the artisan command? It doesn't seem to be handling the stop signal correctly and killing its children.

It could also be an issue with frankenphp's php-cli command trapping signals before they get to the php script. Further investigation is required.

cc: @dunglas
